core/static/core/js/sith-select2.js

```diff
@@ -0,0 +1,257 @@
+ * @property {undefined | function(Object): string} picture_getter
+ *      A callback to get the picture field from the API response
+ * @property {Object | undefined} overrides
+ *      Any other select2 parameter to apply on the config
+ */
+
+/**
+ * @param {Select2Options} options
+ */
+function sithSelect2(options) {
+  const elem = $(options.element);
+  return elem.select2({
+    theme: elem[0].multiple ? "classic" : "default",
+    minimumInputLength: 2,
+    templateResult: select_item_builder(options.picture_getter),
+    ...options.data_source,
+    ...(options.overrides || {}),
+  });
+}
+
+/**
+ * @typedef LocalSourceOptions
+ * @property {undefined | function(): number[]} excluded
+ *        A callback to the ids to exclude from the search
+ */
+
+/**
+ * Build a data source for a Select2 from a local array
+ * @param {Select2Object[]} source The array containing the data
+ * @param {RemoteSourceOptions} options
+ */
+function local_data_source(source, options) {
+  if (!!options.excluded) {
+    const ids = options.excluded();
+    return { data: source.filter((i) => !ids.includes(i.id)) };
+  }
+  return { data: source };
+}
+
+/**
+ * @typedef RemoteSourceOptions
+ * @property {undefined | function(): number[]} excluded
+ *     A callback to the ids to exclude from the search
+ * @property {undefined | function(): Select2Object} result_converter
+ *     A converter for a value coming from the remote api
+ * @property {undefined | Object} overrides
+ *     Any other select2 parameter to apply on the config
+ */
+
+/**
+ * Build a data source for a Select2 from a remote url
+ * @param {string} source The url of the endpoint
+ * @param {RemoteSourceOptions} options
+ */
+function remote_data_source(source, options) {
+  jQuery.ajaxSettings.traditional = true;
+  let params = {
+    url: source,
+    dataType: "json",
+    cache: true,
+    delay: 250,
+    data: function (params) {
+      return {
+        search: params.term,
+        exclude: [
+          ...(this.val() || []).map((i) => parseInt(i)),
+          ...(options.excluded ? options.excluded() : []),
+        ],
+      };
+    },
+  };
+  if (!!options.result_converter) {
+    params["processResults"] = function (data) {
+      return { results: data.results.map(options.result_converter) };
+    };
+  }
+  if (!!options.overrides) {
+    Object.assign(params, options.overrides);
+  }
+  return { ajax: params };
+}
+
+function item_formatter(user) {
+  if (user.loading) {
+    return user.text;
+  }
+}
+
+/**
+ * Build a function to display the results
+ * @param {null | function(Object):string} picture_getter
+ * @return {function(string): jQuery|HTMLElement}
+ */
+function select_item_builder(picture_getter) {
+  return (item) => {
+    const picture =
+      typeof picture_getter === "function" ? picture_getter(item) : null;
+    const img_html = picture
+      ? `<img src="${picture_getter(item)}" alt="item.text"/>`
```

Comment:
```suggestion
      ? `<img src="${picture_getter(item)}" alt="item.text" onerror="this.src='{{ static('core/img/unknown.jpg') }}'"/>`
```
Un truc comme ça pour gérer les profils qui sont marqués en privés et dont on ne peut pas voir l'image

Reply:
Ou alors, le onerror peut être dans un handler passé en paramètre, histoire de rendre ça plus générique